src/rust/zh.manhuagui/src/parser.rs

```diff
@@ -0,0 +1,329 @@
+		artist: author,
+		description: desc,
+		url,
+		categories: vec![],
+		status: MangaStatus::Ongoing,
+		nsfw: MangaContentRating::Safe,
+		viewer: MangaViewer::Scroll,
+	};
+
+	Ok(_manga)
+}
+
+pub fn get_chapter_list(html: Node) -> Result<Vec<Chapter>> {
+	let mut chapters: Vec<Chapter> = Vec::new();
+	let mut index = 1.0;
+
+	let mut div = html.clone();
+	let hidden = html.html().read().contains("__VIEWSTATE");
+	if hidden {
+		let compressed = html.select("#__VIEWSTATE").attr("value").read();
+		let decompressed =
+			String::from_utf16(&decompress_from_base64(compressed.as_str()).unwrap()).unwrap();
+		div = Node::new_fragment(decompressed.as_bytes()).unwrap();
+	}
+
+	for element in div.select(".chapter-list").array() {
+		let chapt_list_div = element.as_node().unwrap();
+
+		for ul_ref in chapt_list_div.select("ul").array() {
+			let ul = ul_ref.as_node().unwrap();
+
+			for li_ref in ul.select("li").array() {
+				let elem = li_ref.as_node().unwrap();
+
+				let url = elem.select("a").attr("href").read();
+				let id = url.clone().replace("/comic/", "").replace(".html", "");
+				let title = elem.select("a").attr("title").read();
+				let mut ch = title
+					.clone()
+					.replace(['第', '话', '卷'], "")
+					.parse::<f32>()
+					.unwrap_or(index);
+				if title.contains('卷') {
+					ch += 99999.0;
+				}
+
+				let chapter = Chapter {
+					id,
+					title,
+					volume: -1.0,
+					chapter: ch,
+					date_updated: index as f64,
+					scanlator: String::new(),
+					url,
+					lang: String::from("zh"),
+				};
+
+				chapters.push(chapter);
+				index += 1.0;
+			}
+		}
+	}
+
+	Ok(chapters)
+}
+
+pub fn get_page_list(base_url: String) -> Result<Vec<Page>> {
+	let mut pages: Vec<Page> = Vec::new();
+
+	let html = Request::new(base_url.as_str(), HttpMethod::Get).html()?;
+
+	let decoder = Decoder::new(html.html().read());
+	let (path, pages_str) = decoder.decode();
+
+	// let mut index = 0;
+	for (index, str) in pages_str.into_iter().enumerate() {
+		let url = format!("https://i.hamreus.com{}{}", path, str);
+		let encoded_url = helper::encode_uri(&url);
+		let page: Page = Page {
+			index: index.try_into().unwrap(),
+			url: encoded_url,
+			base64: String::new(),
+			text: String::new(),
+		};
+		pages.push(page);
+		// index += 1;
+	}
+
+	Ok(pages)
+}
+
+// FILTER
+
+pub fn get_filtered_url(filters: Vec<Filter>, page: i32, url: &mut String) {
+	let mut is_searching = false;
+	let mut search_string = String::new();
+	url.push_str(BASE_URL);
+
+	let mut region: &str = "all";
+	let mut genre: &str = "all";
+	let mut audience: &str = "all";
+	let mut progress: &str = "all";
+
+	for filter in filters {
+		match filter.kind {
+			FilterType::Title => {
+				if let Ok(filter_value) = filter.value.as_string() {
+					search_string
+						.push_str(encode_uri(&filter_value.read().to_lowercase()).as_str());
+					is_searching = true;
+				}
+			}
+			FilterType::Select => {
+				let index = filter.value.as_int().unwrap() as usize;
```

Comment:
either provide a default value or skip it (`continue`)
```suggestion
				let index = filter.value.as_int().unwrap_or(0) as usize;
```